### DISCUSSION TITLE
Text Editors that support Sidef?

### DISCUSSION THREAD

**Discussion by LamentConfiguration** [trizen/sidef#105] [General]
Firstly, I must say this is an amazing little language, I love how pragmatic it is without feeling "heavyweight" on my first few looks at it.

Are there any editors out there with "out of the box" support for full/proper syntax highlighting for Sidef?

I am currently using the Ruby lexer for textadept as mentioned in https://github.com/trizen/sidef/issues/34
This does feel a bit sub-optimal, but is far from a deal-breaker, is there a preferred editor (things like textadept/notepad++/vim)?

If no, I'll see what I can do to learn how to do this for one of the lightweight/portable editors.

**Comment by trizen** [comment]
Thanks! At the moment, I'm not aware of any text editor that include a lexer for Sidef.

However, using the Ruby lexer for syntax highlighting, works great most of the time.

I'm using [Geany](https://github.com/geany/geany), but any other editor that supports Ruby should work fine.
